src/zx_plot.jl

```diff
@@ -0,0 +1,208 @@
+using ZXCalculus
+using LightGraphs
+using GraphPlot: gplot
+using Colors
+using ZXCalculus: qubit_loc
+using Compose
+
+export ZXplot
+
+function Multigraph2Graph(mg::Multigraph)
+    g = SimpleGraph(nv(mg))
+    vs = vertices(mg)
+    for me in edges(mg)
+        add_edge!(g, searchsortedfirst(vs, src(me)), searchsortedfirst(vs, dst(me)))
+    end
+    # multiplicities = ["$(mul(mg, src(e), dst(e)))" for e in edges(g)]
+    multiplicities = ["×$(mul(mg, vs[src(e)], vs[dst(e)]))" for e in edges(g)]
+    for i = 1:length(multiplicities)
+        if multiplicities[i] == "×1"
+            multiplicities[i] = ""
+        end
+    end
+    return g, multiplicities
+end
+
+ZX2Graph(zxd::ZXDiagram) = Multigraph2Graph(zxd.mg)
+ZX2Graph(zxg::ZXGraph) = Multigraph2Graph(zxg.mg)
+
+function et2color(et::String)
+    et == "" && return colorant"black"
+    et == "×2" && return colorant"blue"
+end
+
+function st2color(S::SpiderType.SType)
+    S == SpiderType.Z && return colorant"green"
+    S == SpiderType.X && return colorant"red"
+    S == SpiderType.H && return colorant"yellow"
+    S == SpiderType.In && return colorant"lightblue"
+    S == SpiderType.Out && return colorant"gray"
+end
+
+ZX2nodefillc(zxd) = [st2color(zxd.st[v]) for v in vertices(zxd.mg)]
+
+function ZX2nodelabel(zxd)
+    nodelabel = String[]
+    for v in vertices(zxd.mg)
+        zxd.st[v] == SpiderType.Z && push!(nodelabel, "[$(v)]\n$(print_phase(zxd.ps[v]))")
+        zxd.st[v] == SpiderType.X && push!(nodelabel, "[$(v)]\n$(print_phase(zxd.ps[v]))")
+        zxd.st[v] == SpiderType.H && push!(nodelabel, "[$(v)]")
+        zxd.st[v] == SpiderType.In && push!(nodelabel, "[$(v)]")
+        zxd.st[v] == SpiderType.Out && push!(nodelabel, "[$(v)]")
+    end
+    return nodelabel
+end
+
+function print_phase(p)
+    if typeof(p) <: Rational
+        return "$(p.num)π/$(p.den)"
+    else
+        return "$p π"
+    end
+end
+
+function layout2locs(zxd::ZXDiagram{T,P}) where {T,P}
+    lo = zxd.layout
+    vs = spiders(zxd)
+    locs = Dict()
+    nqubit = lo.nbits
+    frontier_v = ones(T, nqubit)
+    frontier_locs = ones(nqubit)
+
+    while sum([frontier_v[i] <= length(lo.spider_seq[i]) for i = 1:nqubit]) > 0
+        for q = 1:nqubit
+            if frontier_v[q] <= length(lo.spider_seq[q])
+                v = lo.spider_seq[q][frontier_v[q]]
+                nb = neighbors(zxd, v)
+                if length(nb) <= 2
+                    locs[v] = (Float64(frontier_locs[q]), Float64(q))
```

Comment:
why are we doing this type of conversion here?

Reply:
Because the location should be a Float. Without this, it will be an error because q is an integer.